dataland-e2etests/src/test/kotlin/org/dataland/e2etests/utils/ApiAccessor.kt

```diff
@@ -122,6 +126,8 @@ class ApiAccessor {
                 )
             }
         }
+//        Sleep after upload to ensure that dummy QA is passed and data actually available for further processing
+        if (sleepAfterUpload) Thread.sleep(sleepAfterUploadMs)
```

Comment:
I don't like random sleeps. As this is a test, I don't care too much. But I would prefer if we could introduce some waiting mechanism that waits for the right thing to be true (e.g. that a dataset is QAed), and not just waiting a fixed amount of time. 

What do you think?

Reply:
Function has been added that explicitly checks for each upload item if qa is set to passed and waits and retries until this is the case for all uploaded items.

Reply:
ok